src/offset/local/windows.rs

```diff
@@ -8,132 +8,255 @@
+            let transitions =
+                [Transition::new(dst_transition, tz_info.std_offset, tz_info.dst_offset)];
+            lookup_with_dst_transitions(&transitions, *local)
+        }
+        (None, None) => return LocalResult::Single(tz_info.std_offset),
     }
-    LocalResult::None
-}
-
-fn from_utc_time(utc_time: SYSTEMTIME) -> Result<FixedOffset, Error> {
-    let local_time = utc_to_local_time(&utc_time)?;
-    let utc_secs = system_time_as_unix_seconds(&utc_time)?;
-    let local_secs = system_time_as_unix_seconds(&local_time)?;
-    let offset = (local_secs - utc_secs) as i32;
-    Ok(FixedOffset::east_opt(offset).unwrap())
 }
 
-fn from_local_time(local_time: SYSTEMTIME) -> Result<FixedOffset, Error> {
-    let utc_time = local_to_utc_time(&local_time)?;
-    let utc_secs = system_time_as_unix_seconds(&utc_time)?;
-    let local_secs = system_time_as_unix_seconds(&local_time)?;
-    let offset = (local_secs - utc_secs) as i32;
-    Ok(FixedOffset::east_opt(offset).unwrap())
+// The basis for Windows timezone and DST support has been in place since Windows 2000. It does not
+// allow for complex rules like the IANA timezone database:
+// - A timezone has the same base offset the whole year.
+// - There seem to be either zero or two DST transitions (but we support having just one).
+// - As of Vista(?) only years from 2004 until a few years into the future are supported.
+// - All other years get the base settings, which seem to be that of the current year.
+//
+// These details don't matter much, we just work with the offsets and transition dates Windows
+// returns through `GetTimeZoneInformationForYear` for a particular year.
+struct TzInfo {
+    // Offset from UTC during standard time.
+    std_offset: FixedOffset,
+    // Offset from UTC during daylight saving time.
+    dst_offset: FixedOffset,
+    // Transition from standard time to daylight saving time, given in local standard time.
+    std_transition: Option<NaiveDateTime>,
+    // Transition from daylight saving time to standard time, given in local daylight saving time.
+    dst_transition: Option<NaiveDateTime>,
 }
 
-fn system_time_from_naive_date_time(dt: &NaiveDateTime) -> SYSTEMTIME {
-    SYSTEMTIME {
-        // Valid values: 1601-30827
-        wYear: dt.year() as u16,
-        // Valid values:1-12
-        wMonth: dt.month() as u16,
-        // Valid values: 0-6, starting Sunday.
-        // NOTE: enum returns 1-7, starting Monday, so we are
-        // off here, but this is not currently used in local.
-        wDayOfWeek: dt.weekday() as u16,
-        // Valid values: 1-31
-        wDay: dt.day() as u16,
-        // Valid values: 0-23
-        wHour: dt.hour() as u16,
-        // Valid values: 0-59
-        wMinute: dt.minute() as u16,
-        // Valid values: 0-59
-        wSecond: dt.second() as u16,
-        // Valid values: 0-999
-        wMilliseconds: 0,
+impl TzInfo {
+    fn tz_info_for_year(year: i32) -> Option<TzInfo> {
```

Comment:
As suggested previously, let's call this `for_year()` -- no need to duplicate the type name in the method name.

Reply:
Sorry, only now I get your intention. Yes, that is better.